### ISSUE TITLE
Undefined offset: 2 in markdown.php on line 2488

### ISSUE THREAD

**Issue by pavarnos** [michelf/php-markdown#72]
Hi. Seeing this in the production server logs. Haven't been able to reproduce on my development server yet to narrow down the reasons. Will keep hunting... looks like a problem with a malformed table header underline maybe?

Fragment is 

``` php
            $text .= "<tr>\n";
            foreach ($row_cells as $n => $cell)
                $text .= "  <td$attr[$n]>".$this->runSpanGamut(trim($cell))."</td>\n";
            $text .= "</tr>\n";
```

from 
    function _doTable_callback($matches) {
from Markdown 1.0.1p 13 Jan

line 2488 is 

``` php
        $text .= "  <td$attr[$n]>".$this->runSpanGamut(trim($cell))."</td>\n";
```

looks like $attr[$n] is undefined for $n = 2 in this case.

**Comment by michelf** [comment]
I'm not sure either how to reproduce. If you find the corresponding input, let me know.

**Comment by scito** [comment]
It's easy to reproduce. The notice occurs if the separator line has fewer columns.

```
Col 1 | Col 2 | Col 3
- | -
Col 1 | Col 2 | Col 3

Notice: Undefined offset: 2 in MarkdownExtra_Parser->_doTable_callback() (Zeile 2469 von markdown.php).
Notice: Undefined offset: 2 in MarkdownExtra_Parser->_doTable_callback() (Zeile 2488 von markdown.php).
```

**Comment by tfotis** [comment]
Hi, i encountered this issue also on my development server.
If you have E_STRICT as error_reporting, the error should show up.

**Comment by michelf** [comment]
@scito Thanks. I can indeed reproduce the problem. It'll fix it for the next release.

**Comment by michelf** [comment]
Fixed by 7cb355c91e.
